### PR TITLE
Add basic logging

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/ActionPlanService.kt [hunk 1]
```diff
@@ -20,7 +23,9 @@ class ActionPlanService(
    * Retrieves a Prisoner's [ActionPlan] based on their prison number.
    * Throws [ActionPlanNotFoundException] if the [ActionPlan] cannot be found.
    */
-  fun getActionPlan(prisonNumber: String): ActionPlan =
-    persistenceAdapter.getActionPlan(prisonNumber)
+  fun getActionPlan(prisonNumber: String): ActionPlan {
+    log.debug { "Retrieving Action Plan for prisoner [$prisonNumber]" }
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/ActionPlanService.kt:27]: Minor - should this be `log.info` on the basis that the other 2 are?
Or is there some rationale why this is `debug` ?

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM (minor question re: log levels)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/ActionPlanService.kt (hunk 1 above)
Comment (mattwills-moj-digital) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/ActionPlanService.kt:27]: There is a rational :) I think creating an ActionPlan/Goal is useful information that various people (whoever they may be) might be interested in, especially in the absence of reporting.

Comment (nathanrussell-moj-digital) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/ActionPlanService.kt:27]: Ah, that makes sense 👍

**Pull request merged**